app/src/main/java/chat/rocket/android/helper/OkHttpHelper.java

```diff
@@ -1,21 +1,36 @@
 package chat.rocket.android.helper;
 
 import android.content.Context;
+import android.support.annotation.NonNull;
 import com.facebook.stetho.okhttp3.StethoInterceptor;
 
+import java.io.IOException;
 import java.util.concurrent.TimeUnit;
 import chat.rocket.android.RocketChatCache;
 import chat.rocket.android.api.rest.CookieInterceptor;
 import chat.rocket.android.api.rest.DefaultCookieProvider;
+import okhttp3.Call;
+import okhttp3.Callback;
 import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
 
 /**
  * Helper class for OkHttp client.
  */
 public class OkHttpHelper {
+  private static OkHttpClient defaultHttpClient;
   private static OkHttpClient httpClientForUploadFile;
   private static OkHttpClient httpClientForDownloadFile;
   private static OkHttpClient httpClientForWS;
+  private static String contentType;
```

Comment:
> First time you call getContentType() it will probably return null, since it is a async request.

See the [comment](https://github.com/RocketChat/Rocket.Chat.Android/pull/372#discussion_r130874351) below.

> On a list flinging, there will be dozens of requests overwriting contentType

I agree w/ you. There is a way to avoid it, but it involves involves a disabled support. See [this comment](https://github.com/RocketChat/Rocket.Chat.Android/pull/372#issuecomment-319697445).